hip_data_tools/aws/athena.py

```diff
@@ -339,133 +491,62 @@ def drop_table(self, table_name):
         self.run_query("""DROP TABLE IF EXISTS {}""".format(table_name))
 
 
-def generate_csv_ctas(select_query, destination_table, destination_bucket, destination_key,
-                      partition_fields=''):
-    """
-    Method to generate a CTAS query string for creating csv output
-
-    Args:
-        select_query (string): the query to be used for table generation
-        destination_table (string): name of the new table being created
-        destination_bucket (string): the s3 bucket where the data from select query will be stored
-        destination_key (string): the s3 directory where the data from select query will be stored
-        partition_fields (string): partition field names
-
-    Returns (string): CTAS Query in a string
-
-    """
-    return _get_ctas_statement(destination_bucket, destination_key, destination_table,
-                               partition_fields, select_query, file_format='TEXTFILE')
-
-
-def generate_parquet_ctas(select_query, destination_table, destination_bucket, destination_key,
-                          partition_fields=''):
-    """
-    Method to generate a CTAS query string for creating parquet output
-
-    Args:
-        select_query (string): the query to be used for table generation
-        destination_table (string): name of the new table being created
-        destination_bucket (string): the s3 bucket where the data from select query will be stored
-        destination_key (string): the s3 directory where the data from select query will be stored
-        partition_fields (string): partition field names
+@dataclass
+class AthenaTablePartitionsHandlerSettings:
+    """Athena table partitions handler settings"""
+    athena_settings: AthenaSettings
```

Comment:
Inherit from rather than encapsulate
```suggestion
class AthenaTablePartitionsHandlerSettings(AthenaSettings):
    """Athena table partitions handler settings"""
```